### PR TITLE
Initial changes to support adding multiple nfts in one user interaction

### PR DESCRIPTION
**Pull request by dswilson4**
## Explanation

Relevant issue here: https://github.com/MetaMask/metamask-extension/issues/17696

Currently you can only add a single NFT of a single contract to your MM wallet in one user interaction.  The user currently needs to repeat this flow for as many NFTs as they have from a given contract.

This change proposes a user flow to add multiple NFTs from the same contract to your MM wallet in one user interaction as opposed to N interactions for N NFTs.

On an implementation level, I created two new actions that handle multiple token Ids at once in batch.  One action checks for ownership and the second action checks for the token details of the NFTs the user intends to add.  These are both very similar to the actions that exist that handle single NFT additions, these could potentially be merged such that one function handles both cases.

On the frontend, a new checkbox has been added that toggles settings between adding one NFT or multiple NFTs in one user interaction.  I'm looking for suggestions and improvements for this UX, I'm not certain if there's a better component to use (ie. a dropdown, radio group, etc) or if the input schema should change.  Currently, the form accepts a string of comma separated NFT Ids.

### Before (multiple executions of add NFT user flow)

oldMultiAddNFT

### After

addMultipleNFTs

## Manual Testing Steps

1. Navigate to the NFT tab
2. Click the checkbox under the `Add multiple NFT` title
3. Enter NFT contract address
4. Enter a comma-separated list of NFT ids into the form ie. `1,2`
5. Submit and see multiple NFTs added in one interaction

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by dswilson4** [comment]
@pedronfigueiredo @georgewrmarshall @NidhiKJha @garrettbear

**Comment by dswilson4** [comment]
I have read the CLA Document and I hereby sign the CLA

File: ui/store/actions.ts [hunk 1]
```diff
@@ -2195,6 +2195,46 @@ export function addNftVerifyOwnership(
...
+    if (!dontShowLoadingIndicator) {
+      dispatch(showLoadingIndication());
+    }
+    try {
+      for (const tokenId of tokenIDs) {
+        await submitRequestToBackground('addNftVerifyOwnership', [
+          address,
+          tokenId,
+        ]);
+      }
```
Comment (digiwand) [inline ui/store/actions.ts:2219]: what if we had one action that supported adding one or more NFTs?

File: ui/pages/add-nft/add-nft.js [hunk 2]
```diff
@@ -191,13 +274,21 @@ export default function AddNft() {
...
               placeholder={t('nftTokenIdPlaceholder')}
-              value={tokenId}
+              value={checkMultiNFT ? multipleTokenIds : tokenId}
               onChange={(val) => {
                 validateAndSetTokenId(val);
                 setNftAddFailed(false);
               }}
               tooltipText={t('importNFTTokenIdToolTip')}
             />
+            <Text variant={TextVariant.bodySm}>Add multiple NFT</Text>
```
Comment (digiwand) [inline ui/pages/add-nft/add-nft.js:284]: Screen Shot 2023-04-18 at 2 43 33 PM

padding and alignment should be updated

**Review by digiwand — CHANGES_REQUESTED** [review]
Cool seeing this working! 

left comments above. I wonder if design can weight in on this. 

Some thoughts:
- what if we don't need a checkbox and can leave an instruction to the user that they can add multiple NFTs by using a comma?  
- placeholder text should support new input 
- rather than sending a request for each entry, can the API handle receiving multiple values?

**Pull request closed without merging**